lib/node_modules/@stdlib/stats/incr/nanmvariance/lib/main.js

```diff
@@ -0,0 +1,252 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isPositiveInteger = require( '@stdlib/assert/is-positive-integer' ).isPrimitive;
+var isNumber = require( '@stdlib/assert/is-number' ).isPrimitive;
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var format = require( '@stdlib/string/format' );
+
+
+// MAIN //
+
+/**
+* Returns an accumulator function which incrementally computes a moving unbiased sample variance, ignoring NaN values.
+*
+* @param {PositiveInteger} W - window size
+* @param {number} [mean] - mean value
+* @throws {TypeError} first argument must be a positive integer
+* @throws {TypeError} second argument must be a number
+* @returns {Function} accumulator function
+*
+* @example
+* var accumulator = incrnanmvariance( 3 );
+*
+* var s2 = accumulator();
+* // returns null
+*
+* s2 = accumulator( 2.0 );
+* // returns 0.0
+*
+* s2 = accumulator( NaN );
+* // returns 0.0
+*
+* s2 = accumulator( -5.0 );
+* // returns 24.5
+*
+* s2 = accumulator( 3.0 );
+* // returns 19.0
+*
+* s2 = accumulator( NaN );
+* // returns 19.0
+*
+* s2 = accumulator();
+* // returns 19.0
+*
+* @example
+* var accumulator = incrnanmvariance( 3, -2.0 );
+*/
+function incrnanmvariance( W, mean ) {
+	var delta;
```

Comment:
You shouldn't be inlining the entire implementation of `incrmvariance`. Your implementation should be a wrapper around `incrmvariance` which provides specialized `NaN` handling.